### PR TITLE
dev: add pre-commit hook to format project

### LINKED ISSUE #165: dev: add pre-commit hook to format project
The hook will run `scarb fmt` to ensure that the formatting guidelines are respected

### PR DESCRIPTION
**Pull request by aniketpr01**
## Feature Request

Please check the type of change your PR introduces:

- [ ] Bugfix
- [✔️] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build-related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

Project does not have inbuilt format before pushing new changes

Issue Number: Closes #165

## What is the new behavior?
On commit, pre-commit hook will format the project with "scarb fmt" 

- Added changes to install-hook.sh file for adding pre-commit

## Does this introduce a breaking change?

- [] Yes
- [✔️] No

## Added changes to [`CHANGELOG.md`](../CHANGELOG.md)

- [✔️] Yes
- [ ] No

### CODE REVIEW

File: scripts/install_hook.sh [hunk 1]
```diff
@@ -13,17 +13,38 @@ if [ $? -ne 0 ]; then
...
 
+COMMIT_CONTENT='#!/bin/sh
+
+# Run scarb fmt to format the project.
+scarb fmt
+
+# Check if any files were modified after running scarb fmt.
+changed_files=$(git diff --name-only)
+
+if [ -n "$changed_files" ]; then
```
Comment (enitrat) [inline scripts/install_hook.sh:24]: would be better to directly stage them to include them in the commit

```
COMMIT_CONTENT='#!/bin/sh

# Run scarb fmt to format the project.
scarb fmt
# Check if any files were modified after running scarb fmt.
changed_files=$(git diff --name-only)
if [ -n "$changed_files" ]; then
    echo "The following files were reformatted:"
    echo "$changed_files"
    # Stage the changes.
    git add $changed_files
else
    echo "No files were reformatted."
fi
# Continue with the commit
exit 0
'
```

**Review by enitrat — COMMENTED** [review]
Awesome, thanks !

File: scripts/install_hook.sh (hunk 1 above)
Comment (aniketpr01) [reply scripts/install_hook.sh:24]: Done!

**Review by enitrat — DISMISSED** [review]
LGTM

**Pull request merged**